Mail/Views/New Message/AttachmentsManager.swift

```diff
@@ -0,0 +1,266 @@
+    }
+
+    private func createLocalAttachment(name: String,
+                                       type: UTType?,
+                                       disposition: AttachmentDisposition) async -> Attachment {
+        let name = nameWithExtension(name: name,
+                                     correspondingTo: type)
+        let attachmentUUID = UUID().uuidString
+        let attachment = Attachment(uuid: attachmentUUID,
+                                    partId: "",
+                                    mimeType: type?.preferredMIMEType ?? "application/octet-stream",
+                                    size: 0,
+                                    name: name,
+                                    disposition: disposition)
+        attachmentUploadTasks[attachmentUUID] = AttachmentUploadTask(progress: 0, error: nil)
+
+        let savedAttachment = await addLocalAttachment(attachment: attachment)
+        return savedAttachment
+    }
+
+    private func updateLocalAttachment(url: URL, attachment: Attachment) async -> Attachment {
+        let urlResources = try? url.resourceValues(forKeys: [.typeIdentifierKey, .fileSizeKey])
+        let uti = UTType(urlResources?.typeIdentifier ?? "")
+        let mimeType = uti?.preferredMIMEType ?? attachment.mimeType
+        let size = Int64(urlResources?.fileSize ?? 0)
+
+        let newAttachment = Attachment(uuid: attachment.uuid,
+                                       partId: "",
+                                       mimeType: mimeType,
+                                       size: size,
+                                       name: attachment.name,
+                                       disposition: attachment.disposition)
+
+        await updateAttachment(oldAttachment: attachment, newAttachment: newAttachment)
+        return newAttachment
+    }
+
+    func addDocumentAttachment(urls: [URL]) {
+        Task {
+            await withTaskGroup(of: Void.self) { group in
+                for url in urls {
+                    group.addTask {
+                        let localAttachment = await self.createLocalAttachment(name: url.lastPathComponent,
+                                                                               type: UTType.data,
+                                                                               disposition: .attachment)
+                        let updatedAttachment = await self.updateLocalAttachment(url: url, attachment: localAttachment)
+                        do {
+                            _ = try await self.sendAttachment(url: url, localAttachment: updatedAttachment)
+                        } catch {
+                            DDLogError("Error while creating attachment: \(error.localizedDescription)")
+                            await self.updateAttachmentUploadError(attachment: localAttachment, error: error)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    func addImageAttachment(
```

Comment:
I think I would split this method, the new method would only process one element, it would be async and return a Result<cid, error>.

This way we can almost get away with removing the completion closure.
What do you think of this idea?